### ISSUE TITLE
Future work (August 2026)

### ISSUE THREAD

**Issue by rwood-97** [alan-turing-institute/t0-training#33]
From 2026-07-28 planning session, we said we would try to have an Olmo 7B tool calling model trained end to end (or at least have experience/understanding of each stage) by December 2026.

This is the olmo3 pipeline with the bits we need to go through highlighted:

<img width="998" height="347" alt="Image" src="https://github.com/user-attachments/assets/d152238e-5744-445a-b3cc-4543e988ff89" />

The steps for this are:
1. Pre-train a base model ourselves - note: we use 1x chinchilla optimal whereas olmo3 uses ~9.2x chinchilla optimal
2. Do mid-training - olmo3 uses Dolma 3 Dolmino Mix. est. 
3. Do long context training - olmo3 uses Dolma 3 Longmino Mix.

or 

1. Take pre- [pre-trained base model](https://huggingface.co/allenai/Olmo-3-1025-7B) - note: this includes all 3 steps above, so we loose out on getting experience with these if we go this route.

Then post-train:
1. Think SFT - [data here](https://huggingface.co/datasets/allenai/Dolci-Think-SFT-7B), They train for 2x epochs, with 45.4B tokens, 5e-5 LR, 32K max sequence length and document packing (not padding). See olmo3 paper appendix A.6.1.
2. Preference Finetuning via Direct Preference Optimization (Think DPO) - [data here](https://huggingface.co/datasets/allenai/Dolci-Think-DPO-7B), See paper appendix A.6.2. 
3. Think RLVR - [data here](https://huggingface.co/datasets/allenai/Dolci-Think-RL-7B), See paper appendix A.6.3.
